test/functional/test-ad-cid.js

```diff
@@ -56,6 +58,52 @@ function tests(name) {
         setCookie(window, cidScope, '', Date.now() - 5000);
       });
 
+      describe('unit test', () => {
+        let clock;
+        let element;
+        let adElement;
+        beforeEach(() => {
+          clock = sandbox.useFakeTimers();
+          element = document.createElement('amp-ad');
+          element.setAttribute('type', '_ping_');
+          adElement = {
+            element,
+            win: window,
+          };
+        });
+
+        it('provides cid to ad', () => {
+          clientIdScope['_ping_'] = cidScope;
+          const s = installCidService(window);
+          sandbox.stub(s, 'get', scope => {
+            expect(scope).to.be.equal(cidScope);
+            return Promise.resolve('test123');
+          });
+          return getAdCid(adElement).then(cid => {
+            expect(cid).to.equal('test123');
+          });
+        });
+
+        it('times out', () => {
+          clientIdScope['_ping_'] = cidScope;
+          const s = installCidService(window);
+          sandbox.stub(s, 'get', scope => {
+            expect(scope).to.be.equal(cidScope);
+            return timerFor(window).promise(2000);
+          });
+          const p = getAdCid(adElement).then(cid => {
+            expect(cid).to.be.undefined;
+            expect(Date.now()).to.equal(1000);
+          });
+          clock.tick(999);
+          // Let promises resolve before ticking 1 more ms.
```

Comment:
I'm doing this to show that

- when the promise eventually resolves it is already 1000
- which means the promise did not already resolve at 999.